kiwi/container/setup/base.py

```diff
@@ -145,23 +145,21 @@ def setup_root_console(self):
             }
         )
 
-    def setup_static_device_nodes(self):
+    def setup_device_nodes(self):
         """
         Container device node setup
 
         Without subsystems like udev running in a container it is
         required to provide a set of device nodes to let the
         system in the container function correctly. This is
-        done by syncing the host system nodes to the container.
+        done by bind-mounting /dev into the container.
         That this will also create device nodes which are not
         necessarily present in the container later is a know
-        limitation of this method and considered harmless
+        limitation of this method and considered harmless.
         """
         try:
-            data = DataSync('/dev/', self.root_dir + '/dev/')
-            data.sync_data(
-                options=['-a', '-x', '--devices', '--specials']
-            )
+            mm = MountManager('/dev', self.root_dir + '/dev')
```

Comment:
I think the reason for the sync of the device nodes was that in former times a container had to have some device nodes statically available as part of the container archive.

>Without subsystems like udev running in a container it is  required to provide a set of device nodes to let the
 system in the container function correctly.

This is why we rsynced them. With this change the device nodes are available at build time but inside of the container ```/dev``` will stay empty

If this is no longer an issue your change is just perfect. However if container engines still requires some /dev/
nodes e.g console, tty or similar to be present to be able to create an instance we should not change it

If we stick to the mounting part an umount is missing. I suggest to add a destructor to the ContainerSetupBase class and keep the instance of MountManager in class scope (self.mm)

Reply:
> This is why we rsynced them. With this change the device nodes are available at build time but inside of the container /dev will stay empty

I don't think it is needed on my system, but I don't know about others…

Reply:
I think @davidcassany has more insights on that part. I tested with podman on non privileged containers which just worked. I tested with privileged ones and also with containers that runs systemd inside and they worked for me if I did ```-v /dev:/dev``` but the workload also used loops and created dev nodes, thus /dev had to exist.

I cannot foresee if this breaks existing workloads, really no idea but I would be willing to take the risk since a static copy from the host into the container is also not really a nice solution

The last word is on David though :)

Reply:
> I cannot foresee if this breaks existing workloads, really no idea but I would be willing to take the risk since a static copy from the host into the container is also not really a nice solution

I second this, I am also not capable to see if this will cause issues in some very specific cases. IMHO the expectation from contianer engines is that no /dev is required,  in fact note the podman and docker both include a `--tty` to allocate a pseudo terminal driver on the fly. Also binding hosts /dev with containers /dev should be a valid solution if there is a weird case in which the proposed change causes an issue.

If at some point there is a container requiring some device that was present in builder host that is not present in runtime host I'd consider it a bug on the container design and unrelated to KIWI.

Reply:
so I will remove the static_dev setup completely and that's it?

Reply:
> so I will remove the static_dev setup completely and that's it?

That would be ok for me yes. But please let us make sure that the rootfs tree contains a ```/dev``` directory such that ```-v /dev/:/dev``` stays possible. Thanks

Reply:
when you are doing mounts with docker/podman, it does not matter if directory exists or not. the only case where directory would need to exist is when you do manual mount which I believe is not the case..